back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java

```diff
@@ -74,34 +83,37 @@ public MicrometerRegistry(
 		this.totalLabel = Util.makeMicrometerName(totalLabel);
 		this.logFrequency = logFrequency;
 		this.logger = logger;
-		this.registry.config().meterFilter(
-			new MeterFilter() {
-				@Override
-				public DistributionStatisticConfig configure(
-						final Meter.Id id,
-						final DistributionStatisticConfig config
-						) {
-					if(id.getType().equals(
-							Type.DISTRIBUTION_SUMMARY
-							)) {
-						return DistributionStatisticConfig
-								.builder()
-								.percentilePrecision(
-										pctPrecision
-										)
-								.percentiles(
-										(float)pct1/100.0,
-										(float)pct2/100.0,
-										(float)pct3/100.0
-										)
-								.percentilesHistogram(true)
-								.build()
-								.merge(config);
-					}
-					return config;
+		this.registry.config().meterFilter(createMeterFilter(pctPrecision, pct1, pct2, pct3));
+		this.totalReg.config().meterFilter(createMeterFilter(pctPrecision, pct1, pct2, pct3));
+	}
+	
+	private MeterFilter createMeterFilter(Integer pctPrecision, Integer pct1, Integer pct2, Integer pct3) {
+		return new MeterFilter() {
+			@Override
+			public DistributionStatisticConfig configure(
+					final Meter.Id id,
+					final DistributionStatisticConfig config
+					) {
+				if(id.getType().equals(
+						Type.DISTRIBUTION_SUMMARY
+						)) {
+					return DistributionStatisticConfig
+							.builder()
+							.percentilePrecision(
```

Comment:
Shouldn't you adjust bufferLength and expiry ? 
I guess that also setting minimumExpectedValue and maximumExpectedValue can be useful